### PR TITLE
add capIterations to allow fewer deleteSmallFunctions.

### PR DESCRIPTION
**Pull request by jq**
@deerzq @saurfang

### CODE REVIEW

File: training/src/main/scala/com/airbnb/aerosolve/training/AdditiveModelTrainer.scala [hunk 1]
```diff
@@ -304,7 +318,7 @@ object AdditiveModelTrainer {
     val expCorr = scala.math.exp(corr)
     val loss = scala.math.log(1.0 + 1.0 / expCorr)
     val grad = -label / (1.0 + expCorr)
-    val gradWithLearningRate = grad.toFloat * params.learningRate.toFloat * classWeights(label.toInt)
+    val gradWithLearningRate = grad.toFloat * getLearningRate(label, params) * classWeights(label.toInt)
```
Comment (deerzq) [inline training/src/main/scala/com/airbnb/aerosolve/training/AdditiveModelTrainer.scala]: `classWeights` can be used to adjust positive, negative sample importance or learning rate

File: training/src/main/scala/com/airbnb/aerosolve/training/AdditiveModelTrainer.scala [hunk 2]
```diff
@@ -102,6 +104,11 @@ object AdditiveModelTrainer {
         sc.accumulator(0))
       val modelBC = sc.broadcast(model)
       val newModel = sgdTrain(transformed, sgdParams, modelBC)
+
+      if ((i % params.capIterations) == 0) {
```
Comment (deerzq) [inline training/src/main/scala/com/airbnb/aerosolve/training/AdditiveModelTrainer.scala]: so this implies that we delete small functions every `capIterations`?

**Comment by deerzq** [comment]
LGTM
